Tests/NIOSSLTests/UnwrappingTests.swift

```diff
@@ -482,14 +482,9 @@ final class UnwrappingTests: XCTestCase {
         // We haven't spun the event loop, so the handlers are still in the pipeline. Now attempt to unwrap.
         let stopPromise: EventLoopPromise<Void> = clientChannel.eventLoop.makePromise()
         clientHandler.stopTLS(promise: stopPromise)
-
-        do {
-            try stopPromise.futureResult.wait()
-            XCTFail("Did not throw")
-        } catch NIOTLSUnwrappingError.alreadyClosed {
-            // expected
-        } catch {
-            XCTFail("Unexpected error: \(error)")
+        
+        XCTAssertThrowsError(try stopPromise.futureResult.wait()) { error in
+            XCTAssertEqual(.alreadyClosed, error as? NIOTLSUnwrappingError)
```

Comment:
It's possible the error you saw here is an issue with optional promotion? Try `.some(.alreadyClosed))`.